blazorApp/Program.cs

```diff
@@ -1,11 +1,19 @@
 using Microsoft.AspNetCore.Components.Web;
 using Microsoft.AspNetCore.Components.WebAssembly.Hosting;
 using blazorApp;
+using blazorApp.Services;
 
 var builder = WebAssemblyHostBuilder.CreateDefault(args);
 builder.RootComponents.Add<App>("#app");
 builder.RootComponents.Add<HeadOutlet>("head::after");
 
-builder.Services.AddScoped(sp => new HttpClient { BaseAddress = new Uri(builder.HostEnvironment.BaseAddress) });
+builder.Services.AddScoped<HttpClient>();

```

Comment:
you avoid using .baseAddress now...likely due to "+" problem. But if somebody changes their base href, shouldn't the http client use it?
or should we just set this to something via asWebViewUrl? so that other requests work without using asWebViewUrl?

Reply:
I'm definitely open to changing this - I'll explain my reasoning for leaving the `BaseAddress` untouched, and then we can decide if there's a good default value to assign to it.

---

Fundamentally, the "base href" is just the base URL used to resolve relative URLs. However, there are really two uses for relative URLs:
1. Linking to a static asset
2. Linking to another page on the site

These two uses are at odds with each other in the context of a VSCode WebView. If you want to reference a web resource, then the base href should be VSCode's "local resource root", i.e. the root with an external origin where the resources are actually stored.

But if you want to reference another page in the app (e.g. I have an `<a href="/some-page">`), then the base href should be in terms of the document's origin. Otherwise, clicking the link will be treated as an external navigation, and a new browser tab will opened pointing to a resource that doesn't exist.

In order to not make assumptions about how we're using the `HttpClient`, I initially decided to leave the base address empty. However, I think in the vast majority of (and maybe all?) cases, `HttpClient` is going to be used to fetch a web resource. I can't think of a scenario where you'd actually want the base address to be relative to the document's origin, given you can't fetch anything useful other than `index.html` from there.

---

So, I propose that we go with your second suggestion and change the `BaseAddress` to be VSCode's "local resource root", so the default behavior is option 1 mentioned above (but only when using `HttpClient`, of course). It should be noted that even with this change, if you have an HTML element defined in a `.razor` file whose `href` refers to a static asset (e.g. consider an `<img>` element), you will still need to wrap the `href` value in a call to `VsCode.AsWebViewUri()`.

Does this sound good to you?

Reply:
ok, so there is a ResourceRoot and a NavigationRoot.
Not sure where I'd rather have people use VsCode.AsWebViewUri()?? 

I think I lean towards NavigationRoot being the base href.
And making people use VsCode.AsWebViewResourceFileUri() (is that a better name)

Reply:
> I think I lean towards NavigationRoot being the base href

Agreed.

> And making people use VsCode.AsWebViewResourceFileUri() (is that a better name)

I like that name!

We can still change `HttpClient.BaseAddress` to the "ResourceRoot" without changing the base href, and this would reduce the number of cases where calling `VsCode.AsWebViewResourceFileUri()` is necessary (for example, in `FetchData.razor`). Do you agree that's a good change to make? It won't have any negative side effects AFAICT.

Reply:
sounds great. can you please update the PR, and I'll rereview.